iota/filters.py

```diff
@@ -142,3 +143,67 @@ def _apply(self, value):
           'result_type': self.result_type.__name__,
         },
       )
+
+
+class AddressNoChecksum(Trytes):
+  """
+  Validates a sequence as a sequence of trytes.
+  """
+
+  def _apply(self, value):
+    # noinspection PyTypeChecker
+    value =\
+      self._filter(
+        filter_chain = f.Type((binary_type, bytearray, text_type, TryteString)),
+        value = value,
+      ) # type: TrytesCompatible
+
+    if self._has_errors:
+      return None
+
+    # If the incoming value already is an Address then we just make sure it has no checksum
+    if isinstance(value, Address):
+      if value.checksum and not value.is_checksum_valid():
+        return self._invalid_value(
+        value     = value,
+        reason    = self.ADDRESS_BAD_CHECKSUM,
+        exc_info  = True,
+
+        template_vars = {
+          'result_type': self.ADDRESS_BAD_CHECKSUM,
+        },
```

Comment:
❔ I'm not sure that `result_type` is necessary here.  `template_vars` contains values that will be used to format the template string.  For an example, see the `Min` filter:

- https://github.com/eflglobal/filters/blob/1.3.2/filters/number.py#L223
- https://github.com/eflglobal/filters/blob/1.3.2/filters/number.py#L272-L273

In this case, there are no placeholders in the template, so `template_vars` is probably not needed.

That said, there is a `context` argument where we can attach extra values to the resulting error, to help the developer troubleshoot the error.  For example, we could do this:

```python
return self._invalid_value(
  ...
  context = {
    'expected_checksum': value.with_valid_checksum().checksum,
  }
)
```